### PR TITLE
refactor: Optimise jest rules to best practices

### LINKED ISSUE #2028: Optimise jest rules to best practices
Simplify our existing tests to be more readable using jest best practices.

e.g. using https://github.com/sapegin/jest-cheat-sheet

Examples of shortcuts we're not using:

```
.toThrow()
.toBeNull()
```

### PR DESCRIPTION
**Pull request by rishabh1403**
Closes #2028

### CODE REVIEW

**Comment by rishabh1403** [comment]
@rarkins Please Review

File: test/config/presets.spec.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ describe('config/presets', () => {
       const res = await presets.resolveConfigPresets(config);
       expect(res).toMatchSnapshot();
       const rule = res.packageRules[0];
-      expect(rule.automerge).toEqual(true);
+      expect(rule.automerge).toBe(true);
```
Comment (rarkins) [inline test/config/presets.spec.js]: What's the difference? Looks like there's practically no difference?

File: test/platform/bitbucket-server/index.spec.js [hunk 2]
```diff
@@ -225,18 +224,16 @@ describe('platform/bitbucket-server', () => {
           await initRepo();
           api.get.mockClear();
 
-          expect(
-            await bitbucket.ensureComment(5, 'topic', 'content')
-          ).toBeTruthy();
+          expect(await bitbucket.ensureComment(5, 'topic', 'content')).toBe(
+            true
```
Comment (rarkins) [inline test/platform/bitbucket-server/index.spec.js]: Double check that we definitely return true/false and not any string

**Review by rarkins — CHANGES_REQUESTED** [review]

File: test/config/presets.spec.js (hunk 1 above)
Comment (rishabh1403) [reply test/config/presets.spec.js]: There's no difference, changed it to maintain consistency

**Comment by rarkins** [comment]
@rishabh1403 please see my review comments

**Comment by rishabh1403** [comment]
@rarkins manually checked all the files, here's the findings :- 

Uses of `toBeTruthy` in the tests 
- only with `semver.isValid` functions which returns an string/null depending on condition
- `expect(semver.isValid('!= 1.0.0')).toBeTruthy();`

Found no case where `toBeFalsy` would be required,  setting a hard constraint to `toBe(false)` seemed better as all of those returned `false` explicitly

Checked for all the `toBe(true)` cases, all of them return boolean values explicitly

Checked for all the `toBe(false)` cases, all of them return boolean values explicitly

**Comment by rishabh1403** [comment]
@rarkins Please review

**Pull request merged**